### ISSUE TITLE
Node can't sync

### ISSUE THREAD

**Issue by nevata** [ergoplatform/ergo#1386]
![image](https://user-images.githubusercontent.com/34930450/120958689-d05b7100-c78a-11eb-8723-b66cee3f7e4a.png)

**Comment by nevata** [comment]
ergo@wyatt-ubuntu:~/ergo_node$ java -jar ergo-4.0.11.jar --mainnet -c ergo.conf
13:56:26.157 INFO  [main] org.ergoplatform.ErgoApp - Running with args: Args(Some(ergo.conf),Some(MainNet))
13:56:26.217 INFO  [main] o.e.settings.ErgoSettings$ - Running in mainnet network mode
13:56:28.377 INFO  [main] org.ergoplatform.ErgoApp - Working directory: /home/ergo/ergo_node/.ergo
13:56:28.378 INFO  [main] org.ergoplatform.ErgoApp - Secret directory: /home/ergo/ergo_node/.ergo/wallet/keystore
13:56:29.355 ERROR [ctor.default-dispatcher-6] s.c.n.NetworkController - Declared address validation failed:
 /58.247.92.146:9020 not match any of the listening address: Set(/10.10.3.203:9020, /127.0.0.1:9020)
 or Gateway WAN address: None
13:56:29.365 INFO  [ctor.default-dispatcher-6] s.c.n.NetworkController - Declared address: Some(/58.247.92.146:9020)
13:56:29.379 INFO  [ctor.default-dispatcher-8] o.e.mining.ErgoMiner - Trying to use public key from wallet for mining
13:56:29.472 INFO  [tor.default-dispatcher-14] s.c.n.NetworkController - Registering handlers for List((1,GetPeers message), (2,Peers message))
13:56:29.489 INFO  [ctor.default-dispatcher-8] s.c.n.NetworkController - Successfully bound to the port 9020
13:56:29.609 INFO  [tor.default-dispatcher-13] s.c.u.NetworkTimeProvider - New offset adjusted: -29
13:56:29.614 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:29.707 INFO  [tor.default-dispatcher-10] s.c.n.NetworkController - Registering handlers for List((55,Inv), (22,RequestModifier), (33,Modifier), (65,Sync))
13:56:29.851 INFO  [ctor.default-dispatcher-7] scorex.db.LDBFactory$ - Loaded org.fusesource.leveldbjni.JniDBFactory with leveldbjni version 1.8
13:56:30.058 INFO  [ctor.default-dispatcher-7] o.e.n.UtxoNodeViewHolder - History database read
13:56:30.989 INFO  [tor.default-dispatcher-10] o.e.local.CleanupWorker - Cleanup worker started
13:56:31.630 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:32.862 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:32.867 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:32.878 INFO  [tor.default-dispatcher-14] o.e.mining.ErgoMiner - Generating new candidate requested by miner
13:56:32.879 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:33.651 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:34.521 INFO  [ctor.default-dispatcher-6] s.c.n.NetworkController - Connecting to peer: PeerInfo(PeerSpec(unknown,Version(0,0,1),unknown-/159.65.11.55:9030,Some(/159.65.11.55:9030),List()),0,None)
13:56:34.881 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:34.882 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:34.891 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:35.569 INFO  [tor.default-dispatcher-10] s.c.n.NetworkController - Failed to connect to : /159.65.11.55:9030
akka.io.TcpOutgoingConnection$$anon$2: Connect timeout of Some(1 second) expired
13:56:35.711 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:36.940 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:36.941 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:36.942 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:37.731 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
[INFO] [akkaDeadLetter][06/07/2021 13:56:37.901] [ergoref-akka.actor.default-dispatcher-10] [akka://ergoref/deadLetters] Message [akka.pattern.StatusReply] from Actor[akka://ergoref/user/$d#-177754869] to Actor[akka://ergoref/deadLetters] was not delivered. [1] dead letters encountered. If this is not an expected behavior then Actor[akka://ergoref/deadLetters] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
13:56:37.924 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:37.928 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:38.047 INFO  [tor.default-dispatcher-15] o.e.mining.ErgoMiner - Generating new candidate requested by miner
13:56:38.048 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:38.951 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:38.961 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:38.962 WARN  [ergoref-api-dispatcher-9] o.e.n.ErgoReadersHolder - Got GetReaders request in state (None,None,None,None)
13:56:39.532 INFO  [tor.default-dispatcher-12] s.c.n.NetworkController - Connecting to peer: PeerInfo(PeerSpec(unknown,Version(0,0,1),unknown-/209.217.206.254:9030,Some(/209.217.206.254:9030),List()),0,None)
13:56:39.673 INFO  [ctor.default-dispatcher-7] o.e.n.UtxoNodeViewHolder - State and history have the same version a9746bd3d9f4802646a67cdc134112d9d796d074bc3748e09511064c01fe292d, no recovery needed.
13:56:39.674 INFO  [ctor.default-dispatcher-7] o.e.n.UtxoNodeViewHolder - State database read, state synchronized
13:56:39.696 INFO  [ctor.default-dispatcher-7] o.e.n.UtxoNodeViewHolder - Wallet database read
13:56:39.701 INFO  [ergoref-api-dispatcher-9] o.e.n.w.ErgoWalletActor - Initializing wallet actor
13:56:39.763 INFO  [tor.default-dispatcher-15] o.e.n.ErgoNodeViewSynchronizer - 0 persistent modifiers to be downloaded
13:56:39.858 INFO  [ergoref-api-dispatcher-9] o.e.n.w.ErgoWalletServiceImpl - Trying to read wallet in secure mode ..
13:56:39.862 INFO  [ergoref-api-dispatcher-9] o.e.n.w.ErgoWalletServiceImpl - Wallet loaded successfully and locked
13:56:39.869 INFO  [ergoref-api-dispatcher-9] o.e.n.w.ErgoWalletActor - Loading pubkey for miner from cache
13:56:39.870 INFO  [ctor.default-dispatcher-7] o.e.mining.ErgoMiner - Setting public key
13:56:40.553 INFO  [tor.default-dispatcher-10] s.c.n.NetworkController - Failed to connect to : /209.217.206.254:9030
akka.io.TcpOutgoingConnection$$anon$2: Connect timeout of Some(1 second) expired
^C13:56:40.900 WARN  [shutdownHook2] org.ergoplatform.ErgoApp$ - Terminating Actors
13:56:40.921 ERROR [Thread-0] org.ergoplatform.ErgoApp - Unexpected shutdown
13:56:40.921 INFO  [Thread-0] org.ergoplatform.ErgoApp - Stopping network services
13:56:40.923 WARN  [tor.default-dispatcher-10] o.e.n.UtxoNodeViewHolder - Stopping ErgoNodeViewHolder
13:56:40.923 WARN  [tor.default-dispatcher-10] o.e.n.h.s.HistoryStorage - Closing history storage...
13:56:40.924 INFO  [tor.default-dispatcher-13] o.e.mining.ErgoMiner - Stopping miner's threads.
13:56:40.925 WARN  [tor.default-dispatcher-10] o.e.n.state.UtxoState - Closing state's store.
13:56:40.926 WARN  [shutdownHook2] org.ergoplatform.ErgoApp$ - Terminating ActorSystem
13:56:40.932 INFO  [Thread-0] org.ergoplatform.ErgoApp - Stopping actors (incl. block generator)
[INFO] [akkaDeadLetter][06/07/2021 13:56:40.932] [ergoref-akka.actor.default-dispatcher-13] [akka://ergoref/user/networkController] Message [scorex.core.network.NetworkController$ReceivableMessages$ShutdownNetwork$] to Actor[akka://ergoref/user/networkController#1085825553] was not delivered. [2] dead letters encountered. If this is not an expected behavior then Actor[akka://ergoref/user/networkController#1085825553] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [akkaDeadLetter][06/07/2021 13:56:40.974] [ergoref-akka.actor.default-dispatcher-10] [akka://ergoref/user/$c] Message [org.ergoplatform.nodeView.ErgoReadersHolder$GetReaders$] from Actor[akka://ergoref/temp/$a] to Actor[akka://ergoref/user/$c#718867498] was not delivered. [3] dead letters encountered. If this is not an expected behavior then Actor[akka://ergoref/user/$c#718867498] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[WARN] [06/07/2021 13:56:40.980] [Thread-0] [CoordinatedShutdown(akka://ergoref)] Could not addJvmShutdownHook, due to: Shutdown in progress
[INFO] [akkaDeadLetter][06/07/2021 13:56:40.981] [ergoref-akka.actor.default-dispatcher-15] [akka://ergoref/user/$c] Message [org.ergoplatform.nodeView.ErgoReadersHolder$GetReaders$] from Actor[akka://ergoref/temp/$b] to Actor[akka://ergoref/user/$c#718867498] was not delivered. [4] dead letters encountered. If this is not an expected behavior then Actor[akka://ergoref/user/$c#718867498] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [akkaDeadLetter][06/07/2021 13:56:40.981] [ergoref-akka.actor.default-dispatcher-15] [akka://ergoref/user/$c] Message [org.ergoplatform.nodeView.ErgoReadersHolder$GetReaders$] from Actor[akka://ergoref/temp/$c] to Actor[akka://ergoref/user/$c#718867498] was not delivered. [5] dead letters encountered. If this is not an expected behavior then Actor[akka://ergoref/user/$c#718867498] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
13:56:40.999 INFO  [ergoref-api-dispatcher-9] o.e.n.w.ErgoWalletActor - Wallet actor stopped
13:56:41.000 INFO  [tor.default-dispatcher-15] o.e.local.MempoolAuditor - Mempool auditor stopped
13:56:41.066 WARN  [shutdownHook2] org.ergoplatform.ErgoApp$ - Application has been terminated.
13:56:41.066 INFO  [ctor.default-dispatcher-6] org.ergoplatform.ErgoApp - Exiting from the app...

**Comment by nevata** [comment]
ergo {
  node {
    mining = true
  }
}

scorex {
  ntp {
    server = "ntp.aliyun.com"
  }
  network {
    nodeName = "my-ergo-mainnet-node"
    bindAddress = "0.0.0.0:9020"
    declaredAddress = "58.247.92.146:9020"
  }
  restApi {
    apiKeyHash = ""
    bindAddress = "0.0.0.0:9053"
  }
}

**Comment by pragmaxim** [comment]
@nevata Can you pls try with `4.0.12` ? Syncing was improved in this release.

**Comment by nevata** [comment]
I tried it, but the problem persists. Every time this problem occurs, I can only delete it and resynchronize.

**Comment by kushti** [comment]
@nevata so the node being stuck at some height? The log only shows how node suddenly terminated (via Ctrl+C) soon after start. Do you have complete logs ?

**Comment by pragmaxim** [comment]
@kushti I believe that the Wallet digest is updated through the recursive `ScanOnChain / ScanInThePast` actor message in `ErgoWalletActor` ... we should look for WARN log message starting with `Wallet: block in the past reported` in the logs. That might cause this

**Comment by nevata** [comment]
Yes, the node is stuck at a certain altitude and no longer updates and synchronizes. There is no error log output. I think the key to the problem lies in the synchronization state. Why is the altitude different? The node is also displayed as synchronized. 
[ergo.log.zip](https://github.com/ergoplatform/ergo/files/6683329/ergo.log.zip)

**Comment by pragmaxim** [comment]
I will try to propagate the reason/error to the UI : https://github.com/ergoplatform/ergo/issues/1398

**Comment by glasgowm148** [comment]
Please open a new issue with details if the issue persists after #1398

**Comment by ccgarant** [comment]
I'm having this issue now after trying to do port forwarding and set my down stream router as the primary. My rasp pi picked up a new IP address and now i'm getting No Node Found:

[ergoref-api-dispatcher-7] o.e.n.ErgoReadersHolder - Readers are not initialized yet : (None,None,None,None)
